### PR TITLE
Fix parameters unique names

### PR DESCRIPTION
**Pull request by QRemy**
Attempt to fix parameters unique names.

### CODE REVIEW

**Review by Astro-Kirsty — APPROVED** [review]
Thanks @QRemy. This is a really nice addition

**Review by cgalelli — APPROVED** [review]
Thanks @QRemy very nice change

File: gammapy/modeling/models/cube.py [hunk 1]
```diff
@@ -703,6 +719,17 @@ def parameters(self):
             parameters.append(self.spatial_model.parameters)
```
Comment (registerrier) [inline gammapy/modeling/models/cube.py:719]: technically if you introduce the `_models` property, you could simplify this too.

**Review by registerrier — COMMENTED** [review]
Thanks @QRemy. 
Could there be a way to avoid the code duplication of the parameter_unique_names mechanism? 
Does it require an adaptation in the API tutorial or in the user guide, to explain how the parameter names are built?

File: gammapy/modeling/models/cube.py [hunk 2]
```diff
@@ -689,6 +700,11 @@ def spatial_model(self):
         """Spatial norm model."""
         return self._spatial_model
 
+    @property
+    def _models(self):
+        models = self.spectral_model, self.spatial_model
```
Comment (AtreyeeS) [inline gammapy/modeling/models/cube.py:705]: needs the temporal model as well

File: gammapy/modeling/tests/test_fit.py [hunk 3]
```diff
@@ -221,7 +221,7 @@ def test_stat_profile():
     dataset.models.parameters["x"].scan_n_values = 3
     result = fit.stat_profile(datasets=[dataset], parameter="x")
 
-    assert_allclose(result["test.model.x_scan"], [0, 2, 4], atol=1e-7)
+    assert_allclose(result["test.x_scan"], [0, 2, 4], atol=1e-7)
```
Comment (AtreyeeS) [inline gammapy/modeling/tests/test_fit.py:224]: what is the naming logic here? does this solve the issue of whether `x` belongs to `model1` or `model2`?

**Review by AtreyeeS — COMMENTED** [review]
Thanks @QRemy 
Can `def parameters_unique_names` reside in `ModelBase` to avoid code duplication?

File: gammapy/modeling/models/cube.py (hunk 2 above)
Comment (QRemy) [reply gammapy/modeling/models/cube.py:705]: not for FoVBackgroundModel

File: gammapy/modeling/models/cube.py (hunk 1 above)
Comment (QRemy) [reply gammapy/modeling/models/cube.py:719]: OK I will see if a larger cleanup is possible after all the covariance related PRs are in.

**Comment by QRemy** [comment]
> Thanks @QRemy Can `def parameters_unique_names` reside in `ModelBase` to avoid code duplication?

It is defined in `ModelBase` but has to be refined in for each composite models.

**Pull request merged**